src/components/CommentsSection.jsx

```diff
@@ -19,13 +22,59 @@ const CommentsSection = ({ articleId }) => {
           ...comment,
           user: users[comment.author],
         }));
+        commentsWithUserDetails.sort(
+          (a, b) => new Date(b.created_at) - new Date(a.created_at)
+        );
+
         setComments(commentsWithUserDetails);
         setIsLoading(false);
       })
       .catch((error) => setError("Error fetching comments: " + error.message));
     setIsLoading(false);
   }, [articleId, users]);
 
+  const handleCommentChange = (event) => {
+    setCommentText(event.target.value);
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    if (!commentText.trim()) {
+      setError("Comment cannot be empty.");
+      return;
+    }
+    setError("");
+    setSuccessMessage("");
+    setIsSubmitting(true);
+
+    const commentData = {
+      username: loggedInUser.username,
+      body: commentText,
+    };
+
+    postComment(articleId, commentData)
+      .then((newComment) => {
+        setComments((prevComments) => {
+          const updatedComments = [
+            { ...newComment, user: users[newComment.author] },
+            ...prevComments,
+          ];
+          updatedComments.sort(
+            (a, b) => new Date(b.created_at) - new Date(a.created_at)
```

Comment:
You shouldn't need to sort on the front end if your backend already does this ??? (if the Be doesn't then no problem 🙂  )

Reply:
Good point! My backend at the moment only sorts the articles by date, but after this sprint I will add the option for comments to be also sorted by date on the backend and make changes to this component. :raised_hands: Thank you for your feedback Harrison! :blush: